### ISSUE TITLE
SNOW-1292069: Failed to connect with v1.10.0 using `externalbrowser` authentication

### ISSUE THREAD

**Issue by duclet** [snowflakedb/snowflake-connector-nodejs#812]
Please answer these questions before submitting your issue. 
In order to accurately debug the issue this information is required. Thanks!

1. What version of NodeJS driver are you using?
v1.10.0
   
2. What operating system and processor architecture are you using?
Manjaro Linux
   
3. What version of NodeJS are you using?
(`node --version` and `npm --version`)
Node v20.9.0

4. What are the component versions in the environment (`npm list`)?

5.Server version:* E.g. 1.90.1
You may get the server version by running a query:
```
SELECT CURRENT_VERSION();
```
6. What did you do?

   Upgraded from version v1.9.3 to the new version. I'm not now able to connect anymore using the EXTERNALBROWSER authentication model. It gets the error: `Failed to connect: SAML response is invalid or matching user is not found. Contact your local system administrator and provide the error code - .`

7. What did you expect to see?

   Connection successfully.

8. Can you set logging to DEBUG and collect the logs?

```
{"level":"INFO","message":"[12:47:18.374 PM]: Trying to initialize Easy Logging"}
{"level":"INFO","message":"[12:47:18.382 PM]: No client config file found in default directories"}
{"level":"INFO","message":"[12:47:18.383 PM]: Easy Logging is disabled as no config has been found"}
{"level":"DEBUG","message":"[12:47:18.384 PM]: 300"}
{"level":"DEBUG","message":"[12:47:18.385 PM]: Contacting SF: /session/v1/login-request?*****&warehouse=****&roleName=*****, (0/7)"}
{"level":"TRACE","message":"[12:47:18.386 PM]: Create and add to cache new agent https://*****.snowflakecomputing.com-keepAlive"}
{"level":"DEBUG","message":"[12:47:18.387 PM]: Proxy settings used in requests: none."}
{"level":"TRACE","message":"[12:47:18.387 PM]: CALL POST with timeout 90000: https://*****.snowflakecomputing.com/session/v1/login-request?*****&warehouse=****&roleName=*****"}
{"level":"DEBUG","message":"[12:47:18.387 PM]: Issuing heartbeat call"}
{"level":"DEBUG","message":"[12:47:18.441 PM]: OCSP validation disabled for *****.snowflakecomputing.com"}
{"level":"DEBUG","message":"[12:47:19.483 PM]: Encountered an error when sending the request. Details: {\"name\":\"OperationFailedError\",\"code\":\"390133\",\"message\":\"SAML response is invalid or matching user is not found. Contact your local system administrator and provide the error code - . The administrator can obtain troubleshooting information at https://docs.snowflake.com/en/user-guide/errors-saml#saml-error-codes.\",\"data\":{\"nextAction\":\"RETRY_LOGIN\",\"authnMethod\":\"FEDERATED\"}}"}
{"level":"DEBUG","message":"[12:47:20.939 PM]: Contacting SF: /session/v1/login-request?requestId=*****&warehouse=****&roleName=*****, (1/7)"}
{"level":"TRACE","message":"[12:47:20.940 PM]: Get agent with id: https://*****.snowflakecomputing.com-keepAlive from cache"}
{"level":"TRACE","message":"[12:47:20.940 PM]: CALL POST with timeout 90000: https://*****.snowflakecomputing.com/session/v1/login-request?requestId=*****&warehouse=****&roleName=*****"}
{"level":"DEBUG","message":"[12:47:21.535 PM]: Encountered an error when sending the request. Details: {\"name\":\"OperationFailedError\",\"code\":\"390133\",\"message\":\"SAML response is invalid or matching user is not found. Contact your local system administrator and provide the error code - . The administrator can obtain troubleshooting information at https://docs.snowflake.com/en/user-guide/errors-saml#saml-error-codes.\",\"data\":{\"nextAction\":\"RETRY_LOGIN\",\"authnMethod\":\"FEDERATED\"}}"}
{"level":"DEBUG","message":"[12:47:25.737 PM]: Contacting SF: /session/v1/login-request?requestId=******&warehouse=****&roleName=*****, (2/7)"}
{"level":"TRACE","message":"[12:47:25.738 PM]: Get agent with id: https://*****.snowflakecomputing.com-keepAlive from cache"}
{"level":"TRACE","message":"[12:47:25.738 PM]: CALL POST with timeout 90000: https://*****.snowflakecomputing.com/session/v1/login-request?requestId=******&warehouse=****&roleName=*****"}

```
   
9. What is your Snowflake account identifier, if any? (Optional)

**Comment by sfc-gh-dszmolka** [comment]
hi and thank you for raising this with us. indeed we're aware of a regression introduced with 1.10.0 affecting `externalbrowser` auth, and it's fixed in https://github.com/snowflakedb/snowflake-connector-nodejs/pull/804

if you can , you can test the fix by installing the connector from `main` branch, otherwise it will be released with the next upcoming driver version around end of 2024 April

**Comment by sfc-gh-dszmolka** [comment]
fix released with 2024 March release cycle in v1.10.1

**Comment by duclet** [comment]
Just to confirm that the fix worked.
